### PR TITLE
[LV][NFC] Marked functions as const. Added LLVM_DEBUG.

### PR DESCRIPTION
**Pull request by nikolaypanchenko**

### CODE REVIEW

**Comment by nikolaypanchenko** [comment]
NFC changes of #96544

File: llvm/lib/Transforms/Vectorize/LoopVectorize.cpp [hunk 1]
```diff
@@ -1431,29 +1431,40 @@ class LoopVectorizationCostModel {
...
     // iteration in scalar form.
-    if (TheLoop->getExitingBlock() != TheLoop->getLoopLatch())
+    if (TheLoop->getExitingBlock() != TheLoop->getLoopLatch()) {
+      LLVM_DEBUG(
+          dbgs() << "LV: Loop requires scalar epilogue: multiple exits\n");
+      return true;
+    }
+    if (IsVectorizing && InterleaveInfo.requiresScalarEpilogue()) {
+      LLVM_DEBUG(dbgs() << "LV: Loop requires scalar epilogue: vectorizable "
+                           "interleaved group\n");
```
Comment (alexey-bataev) [inline llvm/lib/Transforms/Vectorize/LoopVectorize.cpp:1463]: vectorizable? Better explicitly mention that interleaved group requires scalar epilogue

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Comment by fhahn** [comment]
With this we now print the debug messages about requiring a scalar epilogue at multiple places. Would it be possible to compute it once and set a flag indicating whether a scalar epilogue is required to avoid this?

**Pull request merged**